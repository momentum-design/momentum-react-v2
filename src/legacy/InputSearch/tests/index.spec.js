import React from 'react';
import { mount } from 'enzyme';
import { InputSearch } from '@momentum-ui/react-collaboration';

describe('tests for <InputSearch />', () => {
  it('should match normal SnapShot', () => {
    const container = mount(<InputSearch id="1" name="test" />);

    expect(container).toMatchSnapshot();
  });

  it('should match pill SnapShot', () => {
    const container = mount(<InputSearch id="1" name="test" shape="pill" />);

    expect(container).toMatchSnapshot();
  });

  it('should match loading SnapShot', () => {
    const container = mount(<InputSearch id="1" name="test" isLoading />);

    expect(container).toMatchSnapshot();
  });

  it('should pass isLoading prop', () => {
    const container = mount(<InputSearch id="1" name="test" isLoading />);

    expect(container.find('.md-loading-spinner-wrapper').exists()).toEqual(true);
  });

  it('should render Icon component', () => {
    const container = mount(<InputSearch id="1" name="test" shape="pill" />);
    expect(container.find('.md-icon').length).toEqual(1);
  });

  it('should render one Input', () => {
    const container = mount(<InputSearch id="1" name="test" />);

    expect(container.find('input').length).toEqual(1);
    expect(container.children().length).toEqual(1);
  });

  it('should pass disabled attribute', () => {
    const container = mount(<InputSearch id="test123" name="test" disabled />);

    expect(container.find('input').props().disabled).toEqual(true);
  });

  it('should pass readOnly attribute', () => {
    const container = mount(<InputSearch id="test123" name="test" readOnly />);

    expect(container.find('input').props().readOnly).toEqual(true);
  });

  it('should pass value attribute', () => {
    const handleChange = jest.fn();
    const container = mount(
      <InputSearch id="test123" name="test" value="testing" onChange={handleChange} />
    );

    expect(container.find('input').props().value).toEqual('testing');
  });
});
