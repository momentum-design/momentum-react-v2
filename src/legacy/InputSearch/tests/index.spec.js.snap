// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests for <InputSearch /> should match loading SnapShot 1`] = `
<SearchInput
  id="1"
  isLoading={true}
  name="test"
>
  <Input
    ariaDescribedBy={null}
    ariaLabel={null}
    className=""
    clear={false}
    clearAriaLabel={null}
    containerSize=""
    defaultValue=""
    disabled={false}
    helpText=""
    htmlId={null}
    id="1"
    inputAfter={null}
    inputBefore={
      <Spinner
        className=""
        color="black"
        percentage={null}
        showCheck={false}
        showPercentage={false}
        size={20}
      />
    }
    inputClassName=""
    inputRef={null}
    inputSize=""
    isFilled={false}
    label=""
    messageArr={[]}
    multiline={false}
    name="test"
    nestedLevel={0}
    onChange={null}
    onDoneEditing={null}
    onFocus={null}
    onKeyDown={null}
    onMouseDown={null}
    placeholder=""
    readOnly={false}
    secondaryLabel=""
    shape=""
    type="text"
    value=""
  >
    <div
      className="md-input-container"
    >
      <div
        className="md-input__wrapper"
      >
        <InputSection
          className=""
          position="before"
        >
          <span
            className="md-input__before"
          >
            <Spinner
              className=""
              color="black"
              percentage={null}
              showCheck={false}
              showPercentage={false}
              size={20}
            >
              <i
                className="md-spinner md-spinner--20 md-spinner--black"
              />
            </Spinner>
          </span>
        </InputSection>
        <input
          className="md-input md-input--before"
          id="1"
          name="test"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onMouseDown={[Function]}
          tabIndex={0}
          type="text"
          value=""
        />
      </div>
    </div>
  </Input>
</SearchInput>
`;

exports[`tests for <InputSearch /> should match normal SnapShot 1`] = `
<SearchInput
  id="1"
  isLoading={false}
  name="test"
>
  <Input
    ariaDescribedBy={null}
    ariaLabel={null}
    className=""
    clear={false}
    clearAriaLabel={null}
    containerSize=""
    defaultValue=""
    disabled={false}
    helpText=""
    htmlId={null}
    id="1"
    inputAfter={null}
    inputBefore={
      <Icon
        append={false}
        ariaLabel={null}
        buttonClassName=""
        buttonProps={null}
        className=""
        color=""
        description=""
        name="search_20"
        onClick={null}
        prepend={false}
        size={null}
        sizeOverride={false}
        style={null}
        title=""
        type=""
      />
    }
    inputClassName=""
    inputRef={null}
    inputSize=""
    isFilled={false}
    label=""
    messageArr={[]}
    multiline={false}
    name="test"
    nestedLevel={0}
    onChange={null}
    onDoneEditing={null}
    onFocus={null}
    onKeyDown={null}
    onMouseDown={null}
    placeholder=""
    readOnly={false}
    secondaryLabel=""
    shape=""
    type="text"
    value=""
  >
    <div
      className="md-input-container"
    >
      <div
        className="md-input__wrapper"
      >
        <InputSection
          className=""
          position="before"
        >
          <span
            className="md-input__before"
          >
            <Icon
              append={false}
              ariaLabel={null}
              buttonClassName=""
              buttonProps={null}
              className=""
              color=""
              description=""
              name="search_20"
              onClick={null}
              prepend={false}
              size={null}
              sizeOverride={false}
              style={null}
              title=""
              type=""
            >
              <i
                aria-label={null}
                className="md-icon icon icon-search_20"
                style={
                  {
                    "fontSize": 20,
                  }
                }
              />
            </Icon>
          </span>
        </InputSection>
        <input
          className="md-input md-input--before"
          id="1"
          name="test"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onMouseDown={[Function]}
          tabIndex={0}
          type="text"
          value=""
        />
      </div>
    </div>
  </Input>
</SearchInput>
`;

exports[`tests for <InputSearch /> should match pill SnapShot 1`] = `
<SearchInput
  id="1"
  isLoading={false}
  name="test"
  shape="pill"
>
  <Input
    ariaDescribedBy={null}
    ariaLabel={null}
    className=""
    clear={false}
    clearAriaLabel={null}
    containerSize=""
    defaultValue=""
    disabled={false}
    helpText=""
    htmlId={null}
    id="1"
    inputAfter={null}
    inputBefore={
      <Icon
        append={false}
        ariaLabel={null}
        buttonClassName=""
        buttonProps={null}
        className=""
        color=""
        description=""
        name="search_20"
        onClick={null}
        prepend={false}
        size={null}
        sizeOverride={false}
        style={null}
        title=""
        type=""
      />
    }
    inputClassName=""
    inputRef={null}
    inputSize=""
    isFilled={false}
    label=""
    messageArr={[]}
    multiline={false}
    name="test"
    nestedLevel={0}
    onChange={null}
    onDoneEditing={null}
    onFocus={null}
    onKeyDown={null}
    onMouseDown={null}
    placeholder=""
    readOnly={false}
    secondaryLabel=""
    shape="pill"
    type="text"
    value=""
  >
    <div
      className="md-input-container"
    >
      <div
        className="md-input__wrapper"
      >
        <InputSection
          className=""
          position="before"
        >
          <span
            className="md-input__before"
          >
            <Icon
              append={false}
              ariaLabel={null}
              buttonClassName=""
              buttonProps={null}
              className=""
              color=""
              description=""
              name="search_20"
              onClick={null}
              prepend={false}
              size={null}
              sizeOverride={false}
              style={null}
              title=""
              type=""
            >
              <i
                aria-label={null}
                className="md-icon icon icon-search_20"
                style={
                  {
                    "fontSize": 20,
                  }
                }
              />
            </Icon>
          </span>
        </InputSection>
        <input
          className="md-input md-input--pill md-input--before"
          id="1"
          name="test"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onMouseDown={[Function]}
          tabIndex={0}
          type="text"
          value=""
        />
      </div>
    </div>
  </Input>
</SearchInput>
`;
